valhalla/sif/dynamiccost.h

```diff
@@ -868,13 +868,38 @@ class DynamicCost {
 };
 
 using cost_ptr_t = std::shared_ptr<DynamicCost>;
+using mode_costing_t = std::array<cost_ptr_t, static_cast<size_t>(TravelMode::kMaxTravelMode)>;
```

Comment:
this is something we used all over the code so i made it an officially sanctioned structure here